model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/client/JpaClientModelCriteriaBuilder.java

```diff
@@ -0,0 +1,131 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.storage.jpa.client;
+
+import org.keycloak.models.map.storage.jpa.client.entity.JpaClientEntity;
+import org.keycloak.models.map.storage.jpa.client.entity.JpaClientAttributeEntity;
+import java.util.Arrays;
+import java.util.function.BiFunction;
+import java.util.stream.Stream;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.Join;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.ClientModel.SearchableFields;
+import org.keycloak.models.map.storage.jpa.hibernate.jsonb.JsonbType;
+import org.keycloak.models.map.storage.CriterionNotSupportedException;
+import org.keycloak.models.map.storage.ModelCriteriaBuilder;
+import org.keycloak.storage.SearchableModelField;
+
+public class JpaClientModelCriteriaBuilder implements ModelCriteriaBuilder<ClientModel, JpaClientModelCriteriaBuilder> {
+
+    private BiFunction<CriteriaBuilder, Root<JpaClientEntity>, Predicate> predicateFunc = null;
+
+    public JpaClientModelCriteriaBuilder() {
+    }
+
+    private JpaClientModelCriteriaBuilder(BiFunction<CriteriaBuilder, Root<JpaClientEntity>, Predicate> predicateFunc) {
+        this.predicateFunc = predicateFunc;
+    }
+
+    private void validateValue(Object[] value, int expectLength, SearchableModelField field, Operator op) {
+        if (value == null || value.length != expectLength) {
+            throw new CriterionNotSupportedException(field, op, "Invalid argument: " + Arrays.toString(value));
+        }
+    }
+
+    @Override
+    public JpaClientModelCriteriaBuilder compare(SearchableModelField<? super ClientModel> modelField, Operator op, Object... value) {
+        switch (op) {
+            case EQ:
+                if (modelField.equals(SearchableFields.REALM_ID) || 
+                    modelField.equals(SearchableFields.CLIENT_ID) ||
+                    modelField.equals(SearchableFields.ENABLED)) {
+
+                    validateValue(value, 1, modelField, op);
+
+                    return new JpaClientModelCriteriaBuilder((cb, root) -> 
+                        cb.equal(root.get(modelField.getName()), value[0])
+                    );
+                } else if (modelField.equals(SearchableFields.SCOPE_MAPPING_ROLE)) {
+                    validateValue(value, 1, modelField, op);
+
+                    return new JpaClientModelCriteriaBuilder((cb, root) -> 
+                        cb.isTrue(cb.function("@>",
+                            Boolean.TYPE,
+                            cb.function("->", JsonbType.class, root.get("metadata"), cb.literal("fScopeMappings")),
+                            cb.literal("\"" + value[0].toString().replace("\"", "\\\"") + "\"")))
```

Comment:
This escaping should be perhaps extracted to a separate method used also from other `literal` invocations upon a JSON string.
Also I'm wondering if this is sufficient. Could we leverage Jackson, e.g.
```java
System.out.println(JsonbType.MAPPER.writeValueAsString("hello\"my friend,\nhow are'you?&amp;"));
```
outputs
```
"hello\"my friend,\nhow are'you?&amp;"
```